jazz_metrics/components/utils.js

```diff
@@ -0,0 +1,224 @@
+
+function massageData(assetResults, eventBody) {
+
+    var output_obj = {};
+    output_obj = {
+        "domain": eventBody.domain,
+        "service": eventBody.service,
+        "environment": eventBody.environment,
+        "end_time": eventBody.end_time,
+        "start_time": eventBody.start_time,
+        "interval": eventBody.interval,
+        "statistics": eventBody.statistics,
+        "assets": assetResults
+    }
+
+    return output_obj;
+};
+
+function assetData (results, assetItem) {
+    var asset_obj = {};
+   
+    asset_obj = {
+        "type": assetItem.type,
+        "asset_name": assetItem.asset_name,
+        "statistics": assetItem.statistics,
+        "metrics": []
+    };
+
+    var metricsArr = [];
+
+    results.forEach(function(key) {
+        var metricsObj = {
+            "metric_name": key.Label,
+            "datapoints": key.Datapoints
+        }
+        metricsArr.push(metricsObj);
+    });
+    asset_obj.metrics = metricsArr;
+    return asset_obj;
+};
+
+function getNameSpaceAndMetriDimensons (nameSpaceFrmAsset){
+    var output_obj = {};
+    output_obj["isError"] = false;
+    var paramMetrics = [];
+    var nameSpace = nameSpaceFrmAsset.toLowerCase();
+    
+    var namespacesList = metricConfig.namespaces;
+
+    var supportedNamespace = namespacesList[nameSpace];
+
+    paramMetrics = supportedNamespace["metrics"];
+
+    var awsAddedNameSpace = nameSpace.indexOf('aws/') === -1 ? 'aws/' + nameSpace : nameSpace;
+    awsAddedNameSpace = awsAddedNameSpace.replace(/ /g, "");
+    var awsNameSpace;
+
+    switch (awsAddedNameSpace) {
+        case 'aws/apigateway':
+            awsNameSpace = 'AWS/ApiGateway';
+            break;
+        case 'aws/lambda':
+            awsNameSpace = 'AWS/Lambda';
+            break;
+        case 'aws/cloudfront':
+            awsNameSpace = 'AWS/CloudFront';
+            break;
+        case 'aws/s3':
+            awsNameSpace = 'AWS/S3';
+            break;
+        default: 
+            output_obj["isError"] = true;
+            output_obj["awsNameSpace"] = "Invalid";
+    }
+    output_obj["paramMetrics"] = paramMetrics;
+    output_obj["awsNameSpace"] = awsNameSpace;
+    return output_obj;
+};
+
+function extractValueFromString (string, keyword){
+
+    var value = "";
+    var startIndex = string.indexOf(keyword +":") + (keyword +":").length;
+    var extractedString = string.substring(startIndex, string.length);
+    var extractedSplitString = extractedString.split(":");
+    var value = extractedSplitString[0];
+    return value;
+};
+
+function getApiName (string){
+    var value;
+    if(Object.keys(global_config.APINAME).indexOf(string) > -1){
+        value = global_config.APINAME[string];
+    } else {
+        value = "*"
+    }
+    return value;
+
+};
+
+function getAssetsObj (assetsArray,userStatistics){
+    var newAssetArr  = [];
+    var namespaces = metricConfig.namespaces;
+
+    assetsArray.forEach(function(asset) {
```

Comment:
too much going on here
-loop within a loop and then more logic inside the loop. please simplify this by extracting out in modules/functions. 
consider using modules for the logic inside switch so that each service can have a module and you can initialize the module using something a factory pattern.

Reply:
Worked as mentioned.